src/linter/block.go

```diff
@@ -1623,17 +1602,22 @@ func (b *BlockWalker) handleIf(s *stmt.If) bool {
 	linksCount := 0
 
 	if s.Stmt != nil {
-		linksCount += walk(s.Stmt)
+		// We evaluate s.Cond inside a new context to avoid
```

Comment:
There are some (rare) cases when it might actually matter. For example:

```
if ($a = do_something()) {
// ...
} else {
// ...
}
```

In this example `$a` will be always defined (in all contexts, including the else block) because it was assigned. It does not mean that all expressions in the first if statement will be executed though but the basic idea why I thought we need to execute the first if condition in the base context was in order to handle this scenario.

Reply:
I like your approach better anyway. It does indeed solve a lot of issues that you mentioned.